cmd/kyma/apply/function/function.go

```diff
@@ -84,19 +84,21 @@ func (c *command) Run() error {
 		return err
 	}
 
-	operators := map[operator.Operator][]operator.Operator{
-		operator.NewGenericOperator(client.Resource(operator.GVKFunction).Namespace(configuration.Namespace), function): {
-			operator.NewTriggersOperator(client.Resource(operator.GVKTriggers).Namespace(configuration.Namespace), triggers...),
-		},
-	}
+	var gitOperator operator.Operator
 
 	if configuration.Source.Type == workspace.SourceTypeGit {
 		gitRepository, err := resources.NewPublicGitRepository(configuration)
 		if err != nil {
 			return errors.Wrap(err, "Unable to read the Git repository from the provided configuration")
 		}
-		gitOperator := operator.NewGenericOperator(client.Resource(operator.GVRGitRepository).Namespace(configuration.Namespace), gitRepository)
-		operators[gitOperator] = nil
+		gitOperator = operator.NewGenericOperator(client.Resource(operator.GVRGitRepository).Namespace(configuration.Namespace), gitRepository)
+	}
+
+	operators := map[operator.Operator][]operator.Operator{
+		gitOperator: {},
```

Comment:
What happens when you have a different type of source? You will have a `nil` key with an empty slice.

I would refactor this code so that the git operator gets added to the map only if the source type is git.

So first create the map with the generic operator that is added in all cases and later add the git one only if necessary.

Reply:
Exactly. When the type of source is git, then there is an initialized operator with no child operators (operator: {}), but in a case with inline function, there is just nil with no child operators (nil: {}. `nil: nil` gives the same result). The manager works fine in all those cases ( [click for more](https://github.com/kyma-incubator/hydroform/blob/master/function/pkg/manager/manager_test.go#L239) ). Of course, I can change it and append the operator to the map when the type of source is set as `git`. But I think this is only a visual change because this will not speed up the code ;)